File: etlAuto/etl/ControlFilter.java
package etl;

import java.io.File;
import java.io.FilenameFilter;

class ControlFilter
  implements FilenameFilter
{
  static final int RCV_FLAG = 1;
  static final int MASTER_FLAG = 2;
  static final int MESSAGE_FLAG = 3;
  static final int LOG_FLAG = 4;
  private int rcv_master_flag;

  public ControlFilter(int mode)
  {
    this.rcv_master_flag = mode;
  }

  public boolean accept(File dir, String name) {
    if (name.length() <= 12)
      return false;
    if (this.rcv_master_flag == 1)
    {
      return name.startsWith("dir.");
    }

    if (this.rcv_master_flag == 2)
      if (name.endsWith(".DIR"))
        return true;
    else if (this.rcv_master_flag == 4)
      if (name.endsWith(".log"))
        return true;
    else if ((this.rcv_master_flag == 3) && 
      (name.endsWith(".msg"))) {
      return true;
    }
    return false;
  }
}

File: etlAuto/etl/ControlFilter.py
# -*- coding: utf-8 -*-
import os
import sys
import re
import datetime
import time
import logging

class ControlFilter:
    ## 由于python不具备const常量定义功能，暂时先不实现const变量功能
    RCV_FLAG = 1
    MASTER_FLAG =2
    MESSAGE_FLAG = 3
    LOG_FLAG = 4
    rcv_master_flag = 0
    pattern = ""

    def __init__(self,mode):
        self.rcv_master_flag = mode

    def getPattern(self):
        if(self.rcv_master_flag == self.RCV_FLAG):
            self.pattern = "dir.*"
        if(self.rcv_master_flag == self.MASTER_FLAG):
            self.pattern = "*.DIR"
        if(self.rcv_master_flag == self.MESSAGE_FLAG):
            self.pattern = "*.msg"
        if(self.rcv_master_flag == self.LOG_FLAG):
            self.pattern = "*.log"

        return self.pattern        

if __name__ == "__main__":
    ctrlFilter = ControlFilter(4)
    print ctrlFilter.getPattern()

 
